src/etc/inc/unbound.inc

```diff
@@ -175,6 +175,14 @@ function unbound_generate_config_text($unboundcfg = NULL, $cfgsubdir = "") {
 		$module_config .= 'python ';
 	}
 
+	// Setup DNS64 support
+	if (isset($unboundcfg['dns64'])) {
+		$module_config .= 'dns64 ';
+		if (!empty($unboundcfg['dns64prefix']) && !empty($unboundcfg['dns64netbits'])) {
+			$dns64_conf = 'dns64-prefix: ' . $unboundcfg['dns64prefix'] . '/' . $unboundcfg['dns64netbits'];
```

Comment:
The GUI portion mentions using a default value for the prefix if the user does not specify a prefix value. The linked doc from unbound doesn't say that it assumes one if the directive is missing, so it looks like this needs to have a check if the supplied values are empty/invalid here which then adds the directive with default values instead.

Something like this:

    $dns64_conf = 'dns64-prefix: ';
    if (is_subnetv6($unboundcfg['dns64prefix'] . '/' . $unboundcfg['dns64netbits'])) {
    	$dns64_conf .= gen_subnetv6($unboundcfg['dns64prefix'], $unboundcfg['dns64netbits']);
    } else {
    	$dns64_conf .= "64:ff9b::/96";
    }

Reply:
all fixed